app/renderer/components/Inspector/Screenshot.js

```diff
@@ -34,12 +34,87 @@ export default class Screenshot extends Component {
     });
   }
 
-  async handleScreenshotClick () {
-    const {screenshotInteractionMode, applyClientMethod,
-           swipeStart, swipeEnd, setSwipeStart, setSwipeEnd} = this.props;
-    const {x, y} = this.state;
 
-    if (screenshotInteractionMode === 'tap') {
+  clickScreenshotElement (source, selectedElement, scaleRatio, x, y, highlighterXOffset, highlighterYOffset) {
+    const {selectElement, unselectElement} = this.props;
+    let thisSelectedElementKey = null;
+    let shouldSelectNext = false;
+    let newSelect = true;
+    let isInElement = (element, x, y) => {
+      if (element == null) {
+        return false;
+      }
+      const {x1, y1, x2, y2} = parseCoordinates(element);
+      let left = x1 / scaleRatio + highlighterXOffset;
+      let top = y1 / scaleRatio + highlighterYOffset;
+      let width = (x2 - x1) / scaleRatio;
+      let height = (y2 - y1) / scaleRatio;
+
+      if (x > left && x < left + width && y > top && y < top + height) {
+        return true;
+      }
+
+      return false;
+    };
+
+    if (selectedElement != null) {
+      if (isInElement(selectedElement, x, y)) {
+        newSelect = false;
+      }
+    }
+    let recursive = (element, zIndex = 0) => {
+      if (!element) {
+        return false;
+      }
+
+      let noChildIsInRect = true;
+      for (let i = element.children.length - 1 ; i >= 0 ; i--) {
+        if (recursive(element.children[i], zIndex + 1)) {
+          noChildIsInRect = false;
```

Comment:
Question: When we've discovered that `noChildIsInRect` is false, can we `break` from the for-loop?